oktapam/resource_resource_group.go

```diff
@@ -0,0 +1,172 @@
+package oktapam
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/okta/terraform-provider-oktapam/oktapam/client"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/descriptions"
+)
+
+func resourceResourceGroup() *schema.Resource {
+	return &schema.Resource{
+		Description:   descriptions.ResourceResourceGroup,
+		CreateContext: resourceResourceGroupCreate,
+		ReadContext:   resourceResourceGroupRead,
+		DeleteContext: resourceResourceGroupDelete,
+		UpdateContext: resourceResourceGroupUpdate,
+		Schema: map[string]*schema.Schema{
+			attributes.ID: {
+				Type:     schema.TypeString,
+				Computed: true,
+				// Description is autogenerated
+			},
+			attributes.Name: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions.Name,
+				ValidateDiagFunc: func(i any, p cty.Path) diag.Diagnostics {
+					var diags diag.Diagnostics
+					if s, ok := i.(string); ok {
+						if s == "" {
+							diags = append(diags, diag.Diagnostic{
+								Severity: diag.Error,
+								Summary:  fmt.Sprintf("value for %s must not be empty", attributes.Name),
+							})
+						} else if len(s) > 255 {
+							diags = append(diags, diag.Diagnostic{
+								Severity: diag.Error,
+								Summary:  fmt.Sprintf("value for %s must be between 1 and 255 characters, inclusive", attributes.Name),
+							})
+						} else if !MatchesSimpleName(s) {
+							diags = append(diags, diag.Diagnostic{
+								Severity: diag.Error,
+								Summary:  fmt.Sprintf("value for %s may only contain alphanumeric characters (a-Z, 0-9), hyphens (-), underscores (_), and periods (.)", attributes.Name),
+							})
+						}
+					} else {
+						diags = append(diags, diag.Diagnostic{
+							Severity: diag.Error,
+							Summary:  fmt.Sprintf("value for %s must be a string", attributes.Name),
+						})
+					}
+
+					return diags
+				},
+			},
+			attributes.Description: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions.Description,
+			},
+			attributes.DelegatedResourceAdminGroups: {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Required:    true,
+				Description: descriptions.DelegatedAdminGroups,
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func resourceResourceGroupRead(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	var diags diag.Diagnostics
+	c := m.(client.OktaPAMClient)
+
+	id := d.Id()
+	resourceGroup, err := c.GetResourceGroup(ctx, id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resourceMap := resourceGroup.ToResourceMap()
+	for k, v := range resourceMap {
+		if k == attributes.ID {
```

Comment:
Any purpose to set the Id while reading it back ? Not sure, but there is some difference between setting the id attribute vs setting it as tf resource id.